### PR TITLE
Fix issue 3383, do not allow setting and getting in one command for rspconfig

### PR DESCRIPTION
**Pull request by xuweibj**
#3383 

1. Will not allow setting and getting in one command for rspconfig:
```
# rspconfig cn ip=x.x.x.x netmask
Do not set and get information in the same command for rspconfig.
```

2. For invalid options:
```
# rspconfig cn ipaa
Unsupported command: rspconfig ipaa
```

### CODE REVIEW

**Comment by DengShuaiSimon** [comment]
``CI CHECK RESULT`` : > **PR FORMAT CORRECT**

File: xCAT-server/lib/xcat/plugins/ipmi.pm [hunk 1]
```diff
@@ -8076,11 +8076,37 @@ sub preprocess_request {
         xCAT::Utils->filter_nodes($request, undef, undef, \@bmcnodes, \@nohandle);
         $realnoderange = \@bmcnodes;
     } elsif ($command eq "rspconfig") {
-
         # filter out the nodes which should be handled by ipmi.pm
         my (@bmcnodes, @nohandle);
         xCAT::Utils->filter_nodes($request, undef, undef, \@bmcnodes, \@nohandle);
         $realnoderange = \@bmcnodes;
+
+        if ($realnoderange) {
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/ipmi.pm]: This code can be improved on.  We don't need `optset` and `optget`.  We can use it as a boolean. 

`$optset=0` <-- means we are in get mode
`$optset=1` <--- means we are in set mode 

Then I think we should be able to change the flow so we don't have the same checking option on line 8091 and 8096...

1. Get the option into a variable 
2. check against the variable 

Ex Pseudocode:
``` 
my $optset=0
my $option=$_
if ( = in the string ) { 
    $optset=1
    $option = $1
}
unless ($option =~ .... list .. ) {  
     Unsupported command
     return
}
```

File: xCAT-server/lib/xcat/plugins/ipmi.pm [hunk 2]
```diff
@@ -8076,11 +8076,37 @@ sub preprocess_request {
...
+                } elsif ($_ =~ /^ip$|^netmask$|^gateway$|^vlan$|^userid$|^username$|^password$|^snmpdest|^thermprofile$|^alert$|^garp$|^community$|^backupgateway$/) {
+                    $optget = 1;
+                } else {
+                    $callback->({ errorcode => [1], data => [ "Unsupported command: $command $_"] });
+                    $request = {};
+                    return;
+                }
+            }
+            if ($optset and $optget) {
+                $callback->({ errorcode => [1], data => [ "Do not set and get information in the same command for $command."] });
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/ipmi.pm]: Suggest changing this message to:  

"Usage Error:  Cannot display and change attributes on the same command."

@gurevichmark Any other suggestion for this message?

**Comment by xuweibj** [comment]
Hi @whowutwut , I have modified, please help to review again.

**Comment by xuweibj** [comment]
Hi @whowutwut is there any other comments for this pull request?

**Pull request merged**